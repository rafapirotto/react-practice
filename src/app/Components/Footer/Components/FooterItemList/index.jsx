import React from 'react';

import FooterItem from '../FooterItem';
import { PRIVACY_ROUTE } from '../../../../routes';

const FooterItemList = () => {
  const footerTitles = [
    'FAQ',
    'Help Center',
    'Account',
    'Media Center',
    'Privacy',
    'Jobs',
    'Ways to Watch',
    'Terms of Use',
    'Speed Test',
    'Cookie Preferences',
    'Corporate Information',
    'Contact Us',
  ];

  const renderFooterItems = (titles) =>
    titles.map((title) => {
      let props = {
        title,
        key: title,
      };

      if (title === 'Privacy') {
        props = { ...props, to: PRIVACY_ROUTE };
      }

      return <FooterItem {...props} />;
    });

  return (
    <div
      className="
              mt-12
              grid
              pb-40
              sm:grid-cols-1
              sm:pl-0
              sm:text-center
              md:grid-cols-2
              md:pl-0
              md:text-center
              lg:grid-cols-4
              lg:pl-40
              lg:text-left
          "
    >
      {renderFooterItems(footerTitles)}
    </div>
  );
};

export default FooterItemList;
