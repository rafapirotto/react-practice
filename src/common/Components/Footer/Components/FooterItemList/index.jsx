import React from 'react';

import FooterItem from '../FooterItem';

const FooterItemList = () => {
  const footerTitles = [
    'FAQ',
    'Help Center',
    'Account',
    'Media Center',
    'Privacy',
    'Jobs',
    'Ways to Watch',
    'Terms of Use',
    'Speed Test',
    'Cookie Preferences',
    'Corporate Information',
    'Contact Us',
  ];

  const renderFooterItems = (titles) =>
    titles.map((title) => <FooterItem title={title} key={title} />);

  return (
    <div
      className="
              mt-12
              grid
              pb-40
              sm:grid-cols-1
              sm:pl-0
              sm:text-center
              md:grid-cols-2
              md:pl-0
              md:text-center
              lg:grid-cols-4
              lg:pl-40
              lg:text-left
          "
    >
      {renderFooterItems(footerTitles)}
    </div>
  );
};

export default FooterItemList;
