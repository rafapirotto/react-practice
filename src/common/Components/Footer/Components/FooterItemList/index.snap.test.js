import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import FooterItemList from '.';

test('It matches the snapshot', () => {
  const component = renderer.create(
    <Router>
      <FooterItemList />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
