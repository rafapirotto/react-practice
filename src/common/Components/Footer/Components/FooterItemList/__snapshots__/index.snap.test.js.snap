// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className=" mt-12 grid pb-40 sm:grid-cols-1 sm:pl-0 sm:text-center md:grid-cols-2 md:pl-0 md:text-center lg:grid-cols-4 lg:pl-40 lg:text-left "
>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      FAQ
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Help Center
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Account
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Media Center
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Privacy
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Jobs
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Ways to Watch
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Terms of Use
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Speed Test
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Cookie Preferences
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Corporate Information
    </a>
  </div>
  <div
    className="pb-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Contact Us
    </a>
  </div>
</div>
`;
